internal/core/platform.rs

```diff
@@ -301,3 +303,76 @@ impl WindowEvent {
         }
     }
 }
+
+/// A non-zero size represented in the coordinate space of physical device pixels.
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub struct NonZeroPhysicalSize {
+    /// The width in physical pixels.
+    pub width: NonZeroU32,
+    /// The height in physical pixels;
+    pub height: NonZeroU32,
+}
+
+impl NonZeroPhysicalSize {
+    /// Construct a new physical size from the width and height values, that are assumed to be
+    /// in the physical coordinate space.
+    pub const fn new(width: NonZeroU32, height: NonZeroU32) -> Self {
+        Self { width, height }
+    }
+}
+
+impl TryFrom<PhysicalSize> for NonZeroPhysicalSize {
+    type Error = PlatformError;
```

Comment:
```suggestion
    type Error = core::num::TryFromIntError;
```

Or just 

```suggestion
    type Error = ();
```

I think this is not a PlatformError.

Then we can map the error to a string in the call site as  "Empty size passed to foobar" or something like that.

Reply:
I like the first one - thanks!